presto-main/src/main/java/io/prestosql/security/CatalogAccessControlRule.java

```diff
@@ -24,26 +24,38 @@
 public class CatalogAccessControlRule
 {
     private final boolean allow;
+    private final boolean readOnly;
     private final Optional<Pattern> userRegex;
     private final Optional<Pattern> catalogRegex;
 
     @JsonCreator
     public CatalogAccessControlRule(
             @JsonProperty("allow") boolean allow,
+            @JsonProperty("readOnly") boolean readOnly,
             @JsonProperty("user") Optional<Pattern> userRegex,
             @JsonProperty("catalog") Optional<Pattern> catalogRegex)
     {
         this.allow = allow;
+        this.readOnly = readOnly;
         this.userRegex = requireNonNull(userRegex, "userRegex is null");
         this.catalogRegex = requireNonNull(catalogRegex, "catalogRegex is null");
     }
 
-    public Optional<Boolean> match(String user, String catalog)
+    public Optional<Boolean> matchAllowed(String user, String catalog)
     {
         if (userRegex.map(regex -> regex.matcher(user).matches()).orElse(true) &&
                 catalogRegex.map(regex -> regex.matcher(catalog).matches()).orElse(true)) {
             return Optional.of(allow);
         }
         return Optional.empty();
     }
+
+    public Optional<Boolean> matchReadOnly(String user, String catalog)
```

Comment:
Please combine `matchAllowed` and `matchReadOnly` into single `match(String user, String catalog)` method which returns `Optional<AccessMode>`:
```
enum AccessMode {
    FULL,
    READ_ONLY,
    DENIED,
    /**/;
}
```

Reply:
done